### ISSUE TITLE
Cannot read property 'length' in `calculateLocation`

### ISSUE THREAD

**Issue by deiga** [kucherenko/jscpd#420]
**Describe the bug**
jscpd crashes in the middle of analysis, stacktrace below

**To Reproduce**
1. Run jscpd in our project. This started happening today, we have not yet localized which changes caused this.

**Expected behavior**

**Desktop (please complete the following information):**
 - OS: [macOS]
 - OS Version [10.15.7]
 - NodeJS Version [v14.15.0]
 - jscpd version [3.3.22]

**Additional context**
```
$ jscpd
Clone found (tsx):
 - /app/src/components/RS/SignUpPage/Hubspot/ROW/useScript.tsx [29:51 - 39:24] (10 lines, 82 tokens)
   /app/src/components/RS/SignUpPage/Hubspot/US/useScript.tsx [55:59 - 65:23]

Clone found (tsx):
 - /app/src/components/RS/SignUpPage/Hubspot/ROW/index.tsx [10:13 - 32:25] (22 lines, 118 tokens)
   /app/src/components/RS/SignUpPage/Hubspot/US/index.tsx [14:9 - 36:24]

Clone found (tsx):
 - /app/src/components/RS/SignUpPage/Hubspot/Expired/useScript.tsx [19:20 - 31:55] (12 lines, 114 tokens)
   /app/src/components/RS/SignUpPage/Hubspot/ROW/useScript.tsx [27:58 - 39:51]

Clone found (tsx):
 - /app/src/components/GD/SignUpPage/Hubspot/useScript.tsx [29:57 - 39:47] (10 lines, 86 tokens)
   /app/src/components/RS/SignUpPage/Hubspot/US/useScript.tsx [55:59 - 39:51]

Clone found (tsx):
 - /app/src/components/GD/SignUpPage/Hubspot/index.tsx [12:9 - 34:21] (22 lines, 118 tokens)
   /app/src/components/RS/SignUpPage/Hubspot/US/index.tsx [14:9 - 36:24]

Clone found (tsx):
 - /app/src/components/Articles/PaywallNow/Paywall/PaywallPopup.tsx [30:1 - 41:6] (11 lines, 130 tokens)
   /app/src/components/Articles/MediaNow/MediaViewer/components/Header/Header.tsx [72:1 - 82:7]

Clone found (tsx):
 - /app/src/components/common/Footer/index.tsx [123:17 - 136:47] (13 lines, 97 tokens)
   /app/src/components/common/Footer/index.tsx [44:16 - 57:46]

Clone found (tsx):
 - /app/src/components/common/Footer/index.tsx [189:28 - 204:44] (15 lines, 111 tokens)
   /app/src/components/common/Footer/index.tsx [120:34 - 57:46]

Clone found (tsx):
 - /app/src/components/GD/ExpiredPage/GD.tsx [5:96 - 18:26] (13 lines, 130 tokens)
   /app/src/components/GD/ExpiredPage/SignUp.tsx [11:1 - 23:27]

Clone found (tsx):
 - /app/src/templates/RS/RowSignUpPage.tsx [17:50 - 26:106] (9 lines, 86 tokens)
   /app/src/templates/RS/UsSignUpPage.tsx [55:46 - 64:100]

Clone found (tsx):
 - /app/src/templates/RS/RowSignUpPage.tsx [98:40 - 110:28] (12 lines, 113 tokens)
   /app/src/templates/RS/RowSignUpPage.tsx [72:30 - 84:28]

Clone found (tsx):
 - /app/src/templates/GD/ExpiredSignUp.tsx [11:3 - 18:11] (7 lines, 78 tokens)
   /app/src/templates/GD/SignUpPage.tsx [54:3 - 60:15]

Clone found (tsx):
 - /app/src/templates/GD/ExpiredGD.tsx [10:3 - 19:10] (9 lines, 124 tokens)
   /app/src/templates/GD/SignUpPage.tsx [54:3 - 63:11]

Clone found (tsx):
 - /app/src/templates/GD/DeadlinePage.tsx [45:31 - 50:9] (5 lines, 61 tokens)
   /app/src/components/Articles/ToolTipNow/ToolTipLoader/ToolTipLoader.tsx [5:53 - 10:2]

Clone found (tsx):
 - /app/src/components/SelfAssessment/Timeline.tsx [180:97 - 191:12] (11 lines, 100 tokens)
   /app/src/components/SelfAssessment/Timeline.tsx [84:38 - 95:11]

Clone found (tsx):
 - /app/src/components/SelfAssessment/Timeline.tsx [191:97 - 202:12] (11 lines, 101 tokens)
   /app/src/components/SelfAssessment/Timeline.tsx [103:38 - 114:11]

Clone found (tsx):
 - /app/src/components/SelfAssessment/Timeline.tsx [202:97 - 214:14] (12 lines, 104 tokens)
   /app/src/components/SelfAssessment/Timeline.tsx [118:38 - 130:17]

TypeError: Cannot read property 'length' of undefined
    at calculateLocation (/app/node_modules/@jscpd/tokenizer/dist/tokenize.js:72:59)
    at /app/node_modules/@jscpd/tokenizer/dist/tokenize.js:111:32
    at Array.map (<anonymous>)
    at tokenize (/app/node_modules/@jscpd/tokenizer/dist/tokenize.js:111:10)
    at Object.createTokenMapBasedOnCode (/app/node_modules/@jscpd/tokenizer/dist/tokenize.js:116:20)
    at Tokenizer.generateMaps (/app/node_modules/@jscpd/tokenizer/dist/index.js:12:27)
    at Detector.<anonymous> (/app/node_modules/@jscpd/core/dist/detector.js:33:46)
    at Generator.next (<anonymous>)
    at /app/node_modules/@jscpd/core/dist/detector.js:8:71
    at new Promise (<anonymous>)```

**Comment by kucherenko** [comment]
thank you, will investigate

do you use jscpd for tsx files only?

**Comment by deiga** [comment]
Here's our config,
```
{
  "threshold": 5,
  "reporters": ["html", "console", "xml"],
  "format": "javascript,typescript,jsx,tsx",
  "ignore": ["**/cypress/**", "**/__tests__/**", "**/pages/us/*.tsx"],
  "absolute": true,
  "gitignore": true
}
```

**Comment by deiga** [comment]
I managed to localize what code caused this. We had added a `<script type="text/javascript" src="..."></script>` tag to an tsx file. Changing it to a self-closing tag fixed the issue for us.

**Comment by hata6502** [comment]
https://github.com/kucherenko/jscpd/issues/421

**Comment by kucherenko** [comment]
fixed at jscpd@3.3.24
